### ISSUE TITLE
Error in quoting: Linearity checking failed (not a function type)

### ISSUE THREAD

**Issue by mb64** [idris-lang/Idris2#517]
Here's a minimized example that gives me the error.  I'm using the current HEAD.

```idris
$ cat minimized.idr 
import Language.Reflection

pos : FC
pos = MkFC "somewhere" (0,0) (0,0)

thing : TTImp -> TTImp -> ?
thing f arg = `[ bad : ~(IApp pos f arg) ]
$ idris2 --no-banner minimized.idr 
1/1: Building minimized (minimized.idr)
minimized.idr:7:31--7:34:While processing right hand side of thing at minimized.idr:7:1--8:1:
Linearity checking failed on ?{P:pos:1969}_[] (pat arg : $resolved1492 => pat f : $resolved1492 => $resolved1611 not a function type) at:
7	thing f arg = `[ bad : ~(IApp pos f arg) ]
	                              ^^^

```

I did some snooping in the compiler, and it seems like this is the uncaught case:
<details>
<summary>git diff</summary>
<p>

```diff
diff --git a/src/Core/LinearCheck.idr b/src/Core/LinearCheck.idr
index 29e05c5..a0cef23 100644
--- a/src/Core/LinearCheck.idr
+++ b/src/Core/LinearCheck.idr
@@ -338,6 +338,7 @@ mutual
                       pure (App fc f' aerased,
                             glueBack defs env sc',
                             fused ++ aused)
+                NBind _ _ (PVar _ _ _) _ => throw $ GenericMsg fc "←— here is the error!"
                 NApp _ (NRef _ n) _ =>
                       do Just _ <- lookupCtxtExact n (gamma defs)
                               | _ => throw (UndefinedName fc n)
```

</p>
</details>  

I'm not sure what it's trying to tell me -- what's gone wrong here?

**Comment by edwinb** [comment]
It looks like it's treating `pos` as a pattern variable by the usual rules for types, so perhaps qualifying it will help? There's probably an assumption in linearity checking that doesn't take the possibility of reflected terms into account.

**Comment by gallais** [comment]
Fixed by #1713
